src/core/src/bound_evaluate.cpp

```diff
@@ -281,17 +293,14 @@ bool ov::interval_bound_evaluator(const Node* node,
     auto low_1 = ov::evaluate_lower_bound(node->get_input_source_output(1));
     auto up_0 = ov::evaluate_upper_bound(node->get_input_source_output(0));
     auto up_1 = ov::evaluate_upper_bound(node->get_input_source_output(1));
+    if (!low_0 || !low_1 || !up_0 || !up_1)
+        return false;
 
     std::set<TensorVector, TensorVectorCmp> input_variants = {{low_0, low_1},
                                                               {low_0, up_1},
                                                               {up_0, low_1},
                                                               {up_0, up_1}};
 
```

Comment:
Just a general thought:
As the combinations are known and the goal is to avoid pair duplications when `low_0 == up_0` or `low_1 == up_1`,
usage of std::set with custom comparator and sorting the elements by address seems to be more than needed.

Reply:
I wanted this comparator to be generic and valid for various `TensorVector`'s. I agree, it looks a bit overeager for such narrow usage.